docs-src/cloud/migration.md

```diff
@@ -0,0 +1,88 @@
+- **Migrate Workflow Executions:** There are different approaches for new, running, and completed Workflow Executions.
+  - **New Workflow Executions:** After you’ve updated your Client connection code, Temporal Cloud will automatically be responsible for all new Workflow Executions.
+  - **Running Workflow Executions:** Short-running Workflows can often be drained and then started again on Temporal Cloud. Long-running Workflows that cannot be drained and might require you to implement more code changes to pass the state of the currently running Workflow to Temporal Cloud.
+  - **Completed Workflow Executions:** Completed Workflow Execution History cannot yet be migrated to Temporal Cloud.
+
+### Updating Client connection code in your Workers
+
+Whether you’re self-hosting Temporal or using Temporal Cloud, you manage runtime of your code.
+To migrate your Workflows to Temporal Cloud, you need to change some parameters in the Client connection code, such as the endpoint’s hostname and port number.
+
+The changes needed to direct your Workflow to your Temporal Cloud
+Namespace are only a few lines of code, including:
+
+- [Add your SSL certificate and private key](/cloud/saml-intro) associated with your Namespace.
+- [Create a string](/cloud/namespaces-intro) to reference your cloud endpoint `(namespace.unique_id.tmprl.cloud:port)`.
+- [Connect to Temporal Cloud](/cloud/get-started) with your Client.
+- [Configure tcld, the Cloud CLI](/cloud/tcld/index), with the same address, Namespace, and
+  certificate used to create a Client through code.
+
+### Migrating your Workflow Executions
+
+A Temporal Cluster stores the complete Event History for the entire lifecycle of a
+Workflow Execution.
+To migrate from a self-hosted Temporal Cluster to Temporal Cloud, you need to take into account the current state, Event History, and any future expectations of your Workflow Executions.
+
+**New Workflows are automatically be executed on Temporal Cloud.**
+Once your new code is deployed, all new executions of your Workflows will occur automatically on Temporal Cloud instead of your self-hosted Cluster.
+If you maintain your self-hosted instance, you will still be able to use it to access any previous execution history.
+You can also export JSON from your previous execution history, which you can then import into your own analytics system.
+
+**Running Workflows can either be drained or migrated.**
+If your Workflow can be completed before any compelling event which drives a move to Temporal Cloud, then
+those Workflows can be automatically restarted on Temporal Cloud.
+If your Workflows need to run continuously, then you must migrate Workflows while they are running.
+To accomplish this migration, you must cancel your current Workflow and pass the current state to a new Workflow in Temporal Cloud.
```

Comment:
The words "you must" can be removed without changing the meaning of the sentence.